File: scripts/ralph-opencode/src/utils/logger.ts
/**
 * Logger - Pretty console logging
 */

const levels = {
  debug: 0,
  info: 1,
  warn: 2,
  error: 3,
} as const;

let currentLevel = levels.info;

export function setLevel(level: keyof typeof levels): void {
  currentLevel = levels[level];
}

export const logger = {
  debug: (message: string, ...args: any[]): void => {
    if (currentLevel <= levels.debug) {
      console.log(`\x1b[90m[DEBUG]\x1b[0m ${message}`, ...args);
    }
  },
  
  info: (message: string, ...args: any[]): void => {
    if (currentLevel <= levels.info) {
      console.log(`\x1b[36m[INFO]\x1b[0m ${message}`, ...args);
    }
  },
  
  warn: (message: string, ...args: any[]): void => {
    if (currentLevel <= levels.warn) {
      console.log(`\x1b[33m[WARN]\x1b[0m ${message}`, ...args);
    }
  },
  
  error: (message: string, ...args: any[]): void => {
    if (currentLevel <= levels.error) {
      console.log(`\x1b[31m[ERROR]\x1b[0m ${message}`, ...args);
    }
  },
};

File: internal/logger/logger.go
package logger

import (
	"fmt"
	"io"
	"log"
	"os"
	"strings"
)

type Level int

const (
	LevelDebug Level = iota
	LevelInfo
	LevelWarn
	LevelError
)

var (
	currentLevel = LevelInfo
	logger       = log.New(io.MultiWriter(os.Stderr), "", log.Ltime)
)

func Init(w io.Writer) {
	logger.SetOutput(io.MultiWriter(os.Stderr, w))
	logger.SetFlags(log.Ltime)

	levelStr := strings.ToUpper(os.Getenv("LOG_LEVEL"))
	switch levelStr {
	case "DEBUG":
		currentLevel = LevelDebug
	case "INFO":
		currentLevel = LevelInfo
	case "WARN", "WARNING":
		currentLevel = LevelWarn
	case "ERROR":
		currentLevel = LevelError
	default:
		currentLevel = LevelInfo
	}
}

func SetLevel(level Level) {
	currentLevel = level
}

func Debug(format string, v ...interface{}) {
	if currentLevel <= LevelDebug {
		logger.Printf("[DEBUG] "+format, v...)
	}
}

func Info(format string, v ...interface{}) {
	if currentLevel <= LevelInfo {
		logger.Printf("[INFO] "+format, v...)
	}
}

func Warn(format string, v ...interface{}) {
	if currentLevel <= LevelWarn {
		logger.Printf("[WARN] "+format, v...)
	}
}

func Error(format string, v ...interface{}) {
	if currentLevel <= LevelError {
		logger.Printf("[ERROR] "+format, v...)
	}
}

func Printf(format string, v ...interface{}) {
	logger.Printf(format, v...)
}

func Println(v ...interface{}) {
	logger.Println(v...)
}

func Fatal(v ...interface{}) {
	logger.Fatal(v...)
}

func Fatalf(format string, v ...interface{}) {
	logger.Fatalf(format, v...)
}

func Print(v ...interface{}) {
	logger.Print(v...)
}

func Sprintf(format string, v ...interface{}) string {
	return fmt.Sprintf(format, v...)
}
